exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/MapUtility.java

```diff
@@ -228,11 +228,301 @@ public static void writeToMapFromReader(FieldReader fieldReader, BaseWriter.MapW
           fieldReader.copyAsValue(mapWriter.list(MappifyUtility.fieldValue).list());
           break;
         default:
-          throw new DrillRuntimeException(String.format("kvgen does not support input of type: %s", valueMinorType));
+          throw new DrillRuntimeException(String.format(caller
+              + " does not support input of type: %s", valueMinorType));
       }
     } catch (ClassCastException e) {
       final MaterializedField field = fieldReader.getField();
-      throw new DrillRuntimeException(String.format(TYPE_MISMATCH_ERROR, field.getName(), field.getType()));
+      throw new DrillRuntimeException(String.format(caller + TYPE_MISMATCH_ERROR, field.getName(), field.getType()));
+    }
+  }
+
+  public static void writeToMapFromReader(FieldReader fieldReader, BaseWriter.MapWriter mapWriter,
```

Comment:
Could you please add support for `VARDECIMAL` type, implement `ANY_VALUE` aggregate function for this type and add a test for decimals also?

Reply:
@vvysotskyi I think this could this be added in a follow-up PR; what do you think ?

Reply:
Done. Please take a look.